app/components/NodeSelectPanel/NodeSelectPanel.jsx

```diff
@@ -135,8 +143,18 @@ export default class NodeSelect extends React.Component<Props, State> {
   }
 
   renderNodeList = () => {
-    const { nodes, selectedNode } = this.props
-    if (nodes) {
+    const {
+      nodes,
+      selectedNode,
+      loading,
+      theme,
+      loadNodesData,
+      networkId,
+    } = this.props
+    if (loading && !nodes.length) {
+      return <Loading theme={theme} nobackground />
+    }
+    if (nodes.length) {
```

Comment:
can we simplify the condition?

```suggestion
if (!nodes.length) {
  return loading ? (
    <Loading theme={theme} nobackground />
  ) : (
    <DialogueBox
      icon={<WarningIcon className={styles.warningIcon} />}
      renderText={() => (
        <div>
          Oops! There was an issue retrieving metrics from the network.{" "}
          <a onClick={() => loadNodesData({ networkId })}>Retry?</a>
        </div>
      )}
      className={styles.tokenSalePanelDialogueBox}
    />
  );
}

return ... (other condition)
```